cnd/src/http_api/mod.rs

```diff
@@ -306,15 +306,15 @@ impl<'de> Deserialize<'de> for DialInformation {
 /// to another implementation. This allows us to keep the number of arguments to
 /// HTTP API controllers small and still access all the functionality we need.
 #[derive(Debug)]
-pub struct Dependencies<M, S, A, B, N> {
-    pub metadata_store: Arc<M>,
-    pub state_store: Arc<S>,
+pub struct Dependencies<A, B, N> {
+    pub metadata_store: Arc<InMemoryMetadataStore>,
```

Comment:
📣 This `M` trait bound was here so that once the database is implemented, we could choose between initialising in the `main` our structs with either `InMemoryMetadataStore` or a new implementation `SqliteMetadataStore` (I made up the name).

If now you are forcing the use of `InMemory` down in the library (`http_api` module), then how will you replace with sqlite?

Reply:
You are correct, this was the initial idea behind `MetadataStore`.  During previous work on the database it came to light that this was not going to work and that we would keep the metatadata store in memory always and load it with data from the database.  Same applies to the `StateStore`.

Reply:
If the metadata store would always be in memory then can we just rename `InMemoryMetadataStore` to `MetadataStore`? To me, the `InMemory` qualification means that there is an alternative (because after all, everything is in memory....)